hrms/patches/v14_0/update_payroll_frequency_to_none_if_salary_slip_is_based_on_timesheet.py

```diff
@@ -0,0 +1,13 @@
+import frappe
+
+
+def execute():
+	frappe.reload_doc("payroll", "doctype", "salary_structure")
+
+	salary_strucutres = frappe.get_all(
+		"Salary Structure", filters={"salary_slip_based_on_timesheet": 1}, fields=["name"]
+	)
+
+	if len(salary_strucutres) > 0:
+		for salary_strucutre in salary_strucutres:
+			frappe.db.set_value("Salary Structure", salary_strucutre.name, "payroll_frequency", "")
```

Comment:
1. Check typos: `salary_strucutres` -> `salary_structures`
2. You can pluck `name` field in `get_all` query
3. No need to check for the length. `get_all` returns a list. The for loop will not execute for an empty list.

Reply:
By the way, this can be a simple `frappe.qb.update` query with a where clause. No need to run `get_all` and a `for` loop